data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/accumulator/LocalFileBuffer.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink.accumulator;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.NavigableSet;
+import org.opensearch.dataprepper.plugins.sink.S3SinkConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.services.s3.S3Client;
+
+/**
+ * Implements the local file buffer type.
+ */
+public class LocalFileBuffer implements BufferType {
+
+    private static final Logger LOG = LoggerFactory.getLogger(LocalFileBuffer.class);
+    private S3Client s3Client;
+    private S3SinkConfig s3SinkConfig;
+    private File fileAbsolutePath;
+
+    public LocalFileBuffer() {
+    }
+
+    /**
+     * @param s3Client
+     * @param s3SinkConfig
+     */
+    public LocalFileBuffer(final S3Client s3Client, final S3SinkConfig s3SinkConfig) {
+        this.s3Client = s3Client;
+        this.s3SinkConfig = s3SinkConfig;
+    }
+
+    /**
+     * @param bufferedEventSet
+     * @return boolean
+     * @throws InterruptedException
+     */
+    public boolean localFileAccumulate(final NavigableSet<String> bufferedEventSet) throws InterruptedException {
+        boolean isFileUploadedToS3 = Boolean.FALSE;
+        String s3ObjectFileName = ObjectKey.objectFileName(s3SinkConfig);
+        File file = new File(s3ObjectFileName);
+        try (BufferedWriter eventWriter = new BufferedWriter(new FileWriter(s3ObjectFileName))) {
+            for (String event : bufferedEventSet) {
+                eventWriter.write(event);

```

Comment:
Please provide `event.toJsonString()` here.

Reply:
We are already applying `toJsonString()` jsonCodec parse method before control comes to this file.